### PR TITLE
Display multiple users in intiatedBy field in UI

### PR DESCRIPTION
**Pull request by rekhaswaminathan**
Currently, we are showing only one user info in the Initiated By field. 
With this PR, we can show multiple users in the UI if multiple users have made changes between two snapshots. 

- Checks if backend is sending initiatedByList property. Otherwise just uses initiatedBy. Once we have integrated initiatedByList in detector response, it will start to show multiple users

Screenshots:

If two users made changes:
image

More than two users made changes:
image

Hovering over the field displays the list:
image

### CODE REVIEW

File: AngularApp/projects/diagnostic-data/src/lib/models/changesets.ts [hunk 1]
```diff
@@ -10,4 +10,5 @@ export interface Changes {
     initiatedBy: string;
     originalModel: DiffEditorModel;
     modifiedModel: DiffEditorModel;
+    initiatedByList: any;
```
Comment (yantang-msft) [inline AngularApp/projects/diagnostic-data/src/lib/models/changesets.ts]: why not string[] but any?

File: AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts [hunk 2]
```diff
@@ -432,4 +434,21 @@ export class ChangesetsViewComponent extends DataRenderBaseComponent implements
...
+
+
+   getColumnIndexByName(searchColumnName: string):number {
+        return this.changeSetsColumn.findIndex(column => column.columnName.toLowerCase() === searchColumnName.toLowerCase());
+    }
+
+    getInitiatedByUsers(changeSet: any[]):any[] {
+        let initiatedByListIndex = this.getColumnIndexByName('initiatedByList');
+        let initiatedByIndex = this.getColumnIndexByName('initiatedBy');
+        if(initiatedByListIndex > 0) {
```
Comment (yantang-msft) [inline AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts]: >= 0 to be safe

File: AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts [hunk 3]
```diff
@@ -432,4 +434,21 @@ export class ChangesetsViewComponent extends DataRenderBaseComponent implements
...
+            return changeSet[initiatedByListIndex];
+        }
+        if(changeSet[initiatedByIndex] === null || changeSet[initiatedByIndex] === "") {
```
Comment (yantang-msft) [inline AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts]: and check initiatedByIndex >= 0

Comment (yantang-msft) [reply AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts]: Just use
if(initiatedByIndex < 0 || changeSet[initiatedByIndex]) {
 return ["N/A"];
}
to test falsy value, i.e., null, undefined, ""

File: AngularApp/projects/diagnostic-data/src/lib/diagnostic-data.module.ts [hunk 4]
```diff
@@ -74,7 +74,7 @@ import { ChangesetsViewComponent } from './components/changesets-view/changesets
 import { ChangesViewComponent } from './components/changes-view/changes-view.component';
 import { CustomMaterialModule } from './material-module';
 import { DetectorListAnalysisComponent } from './components/detector-list-analysis/detector-list-analysis.component';
-
+import {PopoverModule} from 'ngx-bootstrap';
```
Comment (yantang-msft) [inline AngularApp/projects/diagnostic-data/src/lib/diagnostic-data.module.ts]: nit: space

File: AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts [hunk 5]
```diff
@@ -41,7 +41,8 @@ export class ChangesetsViewComponent extends DataRenderBaseComponent implements
     timeLineDataSet: DataSet;
     changesTimeline: Timeline;
     changeSetsLocalCopy: {};
-    initiatedBy: string = '';
+    initiatedBy: any[];
```
Comment (yantang-msft) [inline AngularApp/projects/diagnostic-data/src/lib/components/changesets-view/changesets-view.component.ts]: nit:  should it be string[]?

File: AngularApp/projects/diagnostic-data/src/lib/models/changesets.ts (hunk 1 above)
Comment (rekhaswaminathan) [reply AngularApp/projects/diagnostic-data/src/lib/models/changesets.ts]: no specific reason. Will change it to string[]

**Comment by rekhaswaminathan** [comment]
@ShekharGupta1988 I made changes to display multiple users as list instead of popover. Using the bootstrap plugin to display as popover is not working currently as it requires a higher version of bootstrap CSS (4.0.0). Our current project is in 3.4.0. Upgrading to bootstrap 4.0.0 is causing regression in other places. This PR has been open for quite sometime  so I spoke with Yun Jung and she is okay with displaying multiple users as list. We will create a work item to convert to popover.

**Comment by ShekharGupta1988** [comment]
> @ShekharGupta1988 I made changes to display multiple users as list instead of popover. Using the bootstrap plugin to display as popover is not working currently as it requires a higher version of bootstrap CSS (4.0.0). Our current project is in 3.4.0. Upgrading to bootstrap 4.0.0 is causing regression in other places. This PR has been open for quite sometime so I spoke with Yun Jung and she is okay with displaying multiple users as list. We will create a work item to convert to popover.

Makes sense. Thanks Rekha :)

**Pull request merged**